docs/proposals/20220718-workload-update-model.md

```diff
@@ -0,0 +1,142 @@
+
+## Motivation
+
+- In Openyurt, we use a native Kubernetes workload where application updates are processed in bulk by administrators. However, in some scenarios, node-level users also would like to manage application updates.
+- In cloud-edge collaboration scenario, the edge is connected to the cloud through the public network, and the edge nodes may be disconnected from the cloud due to various objective or subjective factors. The update of Workloads (Deployment, DaemonSet, etc.) in the native Kubernetes relies on the connectivity of the edge nodes to the cloud master nodes. Therefore, when the cloud-edge network is disconnected, we need to explore a suitable way to update the application.
+
+### Goals
+
+- Provide OTA update: users can control the daemonset application updates through the API exposed on the edge nodes.
+- Add a new controller: for `ready` nodes, execute the daemosnet application updates directly; for `not-ready` nodes, wait for nodes' status to turn ready before executing the update process.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+- In some node-independent scenarios, such as a user owning a node, for example, on a Tesla car, the owner wants to independently update his application, instead of being controlled by the cluster administrator. At this time, the user can complete the application update process through a two-step process.
+  1. Check whether the latest version of the application is available.
+  2. Confirm the execution of application update.
+
+#### Story 2
+
+- In a openyurt cluster, there are some edge nodes, and some node's status are `not-ready` due to the network, when the administrator updates the application version, the administrator expects that for the `ready` edge nodes, the application can be updated to the latest version smoothly; for the `not-ready` edge nodes, they do not block the update process of other nodes.
+
+### Implementation Details
+
+Note: No modification to Kubernetes, keeping it non-invasive, no modification to the native workload controllers.
+
+#### 1 OTA-Update
+
+<div align="center">
+  <img src="../img/OTA.png" width=50% title="OTA">
+</div>
+
+**Pre-requisites**: Set application's updateStrategy to `type: OnDelete`
+
+1. Users query whether a newer version of the application is available via OTA module's API.
+2. OTA module accesses Kubernetes APIServer through client-go library and get the current application's version.
+3. If update is available, users can choose to apply update. (This is only from the user's perspective)
+4. OTA module removes the selected pod from the current node.（This step is achieved by calling the Kubernetes interface）'
+5. Kubelet on this node will synchronize pods information from Kubernetes APIServer.
+6. Kubelet is responsible for creating the latest version of pod.
+
+<div align="center">
+  <img src="../img/process.png" width=40% title="OTA Process">
+</div>
+
+##### OTA API
```

Comment:
where can the user call this api from ? Node?
If the api called from node,  so why don't we operate the pod or podlist?

Reply:
Users should call these APIs on edge nodes. Perhaps we can discuss this at community meeting tomorrow.